nemo/collections/asr/models/asr_model.py

```diff
@@ -62,6 +63,43 @@ def list_available_models(cls) -> 'List[PretrainedModelInfo]':
         list_of_models = model_utils.resolve_subclass_pretrained_model_info(cls)
         return list_of_models
 
+    def add_auxiliary_losses(self, loss: torch.Tensor, reset_registry: bool = True) -> torch.Tensor:
+        """
+        Utility method to enable calculation of auxiliary losses for ASR training.
+
+        Args:
+            loss: The output loss value prior to addition with auxiliary losses.
+            reset_registry: Bool, whether to reset the AccessMixin registry after adding auxiliary losses.
+
+        Returns:
+            Loss tensor used for back propagation.
+        """
+        # Add adapter auxiliary losses, if registered
+        if AccessMixin.is_access_enabled():
+            registry = AccessMixin.get_module_registry(self)
+            log_dict = {}
+
+            for loss_key, loss_registry in registry.items():
+                # Add auxiliary loss to total loss
+                loss_list = loss_registry['adapter_loss']
+                loss_value = sum(loss_list)
+                loss += loss_value
+
+                # Log current loss name and value
+                keys = loss_key.split(".")
+                key = "/".join(keys)
+                key = "adapter_loss/" + key
+                log_dict[key] = loss_value.detach()
+
+            if len(log_dict) > 0:
+                self.log_dict(log_dict)
+
+        if reset_registry:
+            AccessMixin.reset_registry(self)
```

Comment:
Doable, but risks memory leaks if you aren't aware of the other registries. We should play it safer with global reset - and build the registry back part by part and use it up before the reset. Large chance of memory leaks otherwise.

Reply:
Perhaps we can have both global and local reset functions, with global being the one called at start of step

Reply:
I think this reset should not be global